server/api/utils/singletons/k8s.py

```diff
@@ -198,7 +198,10 @@ def list_crds_paginated(
                     "remaining_item_count"
                 ),
             )
-        logger.debug("Finished listing crds")
+
+        if _continue:
+            # we want to show this message only if we needed pagination
+            logger.debug("Finished listing crds")
```

Comment:
I think it needs to be above, in line 193:
```python
if not _continue:
    logger.debug("Finished listing crds")
    break
```

Because only then you actually finish.
If `_continue` it means you have more crds to list -> not finished.